src/support/generation.c

```diff
@@ -190,8 +190,7 @@ __gen_oldest(WT_SESSION_IMPL *session, int which)
      * the sessions that could have been active when we started our check.
      */
     WT_ORDERED_READ(session_cnt, conn->session_cnt);
-    for (oldest = conn->generations[which] + 1, s = conn->sessions, i = 0; i < session_cnt;
-         ++s, ++i) {
+    for (oldest = conn->generations[which], s = conn->sessions, i = 0; i < session_cnt; ++s, ++i) {
```

Comment:
So this doesn't get missed: now the split code is post-incrementing the global generation, we don't need the `+ 1` here.  New threads entering the current generation should not make `__gen_oldest` go backwards.